src/App.js

```diff
@@ -1,14 +1,134 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm  from './components/TodoComponents/TodoForm';
+
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
-  render() {
-    return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
-      </div>
+    state = {
+        items: [],
+        task: '',
+        error: null,
+        search: '',
+    }
+    
+    addItem = event => {
+        event.preventDefault();
+        if(this.state.task === ''){
+            return this.setState({
+                error: 'You must enter a task!'
+            })
+        }
+        const newItem = {
+            task: this.state.task,
+            id: Date.now(),
+            completed: false,
+            matching: true
+        }
+
+        const tempList = [...this.state.items, newItem];
+        const sortedList = tempList.sort((a,b) => {
+            const taskA = a.task;
```

Comment:
Your code prob makes more sense to you looking at it this way. Just wanted to let you know that I think you could have just done return a.task - b.task and gotten the same results.